### ISSUE TITLE
Upgrade from 3.3.0-14 to 3.4 rc3 General error:  1005

### ISSUE THREAD

**Issue by mabere** [pkp/pkp-lib#9062]
**Describe the bug**
Please tell us what happens, what you expected to happen, and why you think it is a bug in the software.

**To Reproduce**
1. Go to Upgrade page
2. Click on 'Upgrade Open Journal System'
3. Upgrade stop and see the following error: 
A database error has occurred: SQLSTATE[HY000]: General error: 1005 Can't create table `ojs_v33014`.`journals` (errno: 150 "Foreign key constraint is incorrectly formed") (SQL: alter table `journals` add constraint `journals_current_issue_id_foreign` foreign key (`current_issue_id`) references `issues` (`issue_id`) on delete set null), referer: http://localhost/ojs34/index.php/index/install/installUpgrade

**What application are you using?**
 OJS version 3.3.0.14 to 3.4rc3
![Screenshot 2023-06-03 at 20-33-13 Upgrade Open Journal Systems Open Journal Systems](https://github.com/pkp/pkp-lib/assets/29719586/fe53e96c-38fd-4a53-b337-37cae2cfddd2)

**Additional information**
Please add any screenshots, logs or other information we can use to investigate this bug report.

**Comment by jonasraoni** [comment]
Hi @mabere,

Could you please provide the output for these two commands, before starting the upgrade:
```sql
SHOW CREATE TABLE journals;
SHOW CREATE TABLE issues;
```

**Comment by jonasraoni** [comment]
I can't see the end of the `issues` table, you can paste only the text instead of a screenshot.

**Comment by mabere** [comment]
here the full result of table issues:

CREATE TABLE `issues` (
  `issue_id` bigint(20) NOT NULL AUTO_INCREMENT,
  `journal_id` bigint(20) NOT NULL,
  `volume` smallint(6) DEFAULT NULL,
  `number` varchar(40) DEFAULT NULL,
  `year` smallint(6) DEFAULT NULL,
  `published` tinyint(4) NOT NULL DEFAULT 0,
  `current` tinyint(4) NOT NULL DEFAULT 0,
  `date_published` datetime DEFAULT NULL,
  `date_notified` datetime DEFAULT NULL,
  `last_modified` datetime DEFAULT NULL,
  `access_status` tinyint(4) NOT NULL DEFAULT 1,
  `open_access_date` datetime DEFAULT NULL,
  `show_volume` tinyint(4) NOT NULL DEFAULT 0,
  `show_number` tinyint(4) NOT NULL DEFAULT 0,
  `show_year` tinyint(4) NOT NULL DEFAULT 0,
  `show_title` tinyint(4) NOT NULL DEFAULT 0,
  `style_file_name` varchar(90) DEFAULT NULL,
  `original_style_file_name` varchar(255) DEFAULT NULL,
  `url_path` varchar(64) DEFAULT NULL,
  PRIMARY KEY (`issue_id`),
  KEY `issues_journal_id` (`journal_id`),
  KEY `issues_url_path` (`url_path`)
) ENGINE=InnoDB AUTO_INCREMENT=144 DEFAULT CHARSET=utf8

**Comment by mabere** [comment]
1. I got another error after a new upgrade try:
[Sun Jun 04 22:24:20.200117 2023] [php:notice] [pid 248:tid 1900] [client 127.0.0.1:50227] SQLSTATE[23000]: Integrity constraint violation: 1452 Cannot add or update a child row: a foreign key constraint fails (`unbaja_v3314`.`#sql-16a0_117`, CONSTRAINT `user_group_stage_context_id` FOREIGN KEY (`context_id`) REFERENCES `journals` (`journal_id`) ON DELETE CASCADE) (SQL: alter table `user_group_stage` add constraint `user_group_stage_context_id` foreign key (`context_id`) references `journals` (`journal_id`) on delete cascade), referer: https://localhost/ojs34/index.php/index/install/upgrade

SQL command result table user_group_stage:
CREATE TABLE `user_group_stage` (
  `context_id` bigint(20) NOT NULL,
  `user_group_id` bigint(20) NOT NULL,
  `stage_id` bigint(20) NOT NULL,
  UNIQUE KEY `user_group_stage_pkey` (`context_id`,`user_group_id`,`stage_id`),
  KEY `user_group_stage_context_id` (`context_id`),
  KEY `user_group_stage_user_group_id` (`user_group_id`),
  KEY `user_group_stage_stage_id` (`stage_id`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8
3. I got from this link https://pkp.sfu.ca/2023/04/26/3-4-release-testing-program-rc3/.

**Comment by mabere** [comment]
Unfortunately, I had dropped the database of failed upgraded. I make a new database and run upgrade. The upgrade stopped and display the following error:
[Sun Jun 04 22:24:20.200117 2023] [php:notice] [pid 248:tid 1900] [client 127.0.0.1:50227] SQLSTATE[23000]: Integrity constraint violation: 1452 Cannot add or update a child row: 
a foreign key constraint fails 
(`unbaja_v3314`.`#sql-16a0_117`, CONSTRAINT `user_group_stage_context_id` FOREIGN KEY (`context_id`) REFERENCES `journals` (`journal_id`) ON DELETE CASCADE) (SQL: alter table `user_group_stage` add constraint `user_group_stage_context_id` 
foreign key (`context_id`) references `journals` (`journal_id`) on delete cascade), referer: https://localhost/ojs34/index.php/index/install/upgrade.
After running the command based on the table user_group_stage table, get the result:
CREATE TABLE `user_settings` (
  `user_id` bigint(20) NOT NULL,
  `locale` varchar(14) NOT NULL DEFAULT '',
  `setting_name` varchar(255) NOT NULL,
  `assoc_type` bigint(20) DEFAULT 0,
  `assoc_id` bigint(20) DEFAULT 0,
  `setting_value` text DEFAULT NULL,
  `setting_type` varchar(6) NOT NULL,
  UNIQUE KEY `user_settings_pkey` (`user_id`,`locale`,`setting_name`,`assoc_type`,`assoc_id`),
  KEY `user_settings_user_id` (`user_id`),
  KEY `user_settings_locale_setting_name_index` (`setting_name`,`locale`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8 COLLATE=utf8_general_ci

**Comment by jonasraoni** [comment]
Hmm, if you used the same database backup you should see the same problem, I was mostly interested on it 😭

About the new issue that you got, I'm already working on it. There are several fixes here, but they are still under review: https://github.com/pkp/pkp-lib/issues/8929

**Comment by mabere** [comment]
Yes. I use exactly the same backup database. But I will do another upgrade try to see if the error still the same or result in another error. 
Great news for fixes in #8929.

**Comment by jonasraoni** [comment]
Thanks for the help! :)

If you know how to install OJS from GitHub, feel free to try the fixes. Otherwise you'll have to follow the same advises that I gave to the other user (clean the orphan entities manually, apply some fixes, etc).

**Comment by mabere** [comment]
Thanks. I have no knowledge to use github. I will try the fixes on #8929.

**Comment by mabere** [comment]
Hi @jonasraoni. I have another upgrade try and got issues that can be fixed by query found in #8929 (e.g., duplication and conflict with specific to local ).
Another error present:
A database error has occurred: SQLSTATE[42000]: Syntax error or access violation: 1091 Can't DROP INDEX `review_form_responses_pkey`; check that it exists (SQL: alter table `review_form_responses` drop index `review_form_responses_pkey`)

Result from query SHOW CREATE table review_form_responses:
CREATE TABLE `review_form_responses` (
`review_form_element_id` biginit(20) NOT NULL,
`review_id` biginit(20) NOT NULL,
`responses_type` varchar(6) DEFAULT NULL,
`responses_value` text DEFAULT NULL,
KEY `review_form_responses_unique` (`review_form_element_id`,`review_id`)
) ENGINE=InnoBD DEFAULT CHARSET=utf8 COLLATE=utf8_general_ci

Note: This is another upgrade from different ojs installation from previous discussed one.

**Comment by jonasraoni** [comment]
Hmm, the table already has the right index 🤔
I guess you had a problem with the upgrade, fixed, then tried to run the upgrade again, right?

Retrying the upgrade without restoring the database from a backup might work sometimes, but not always, it depends how the migration script was written/how it failed (there are 69 of such migration scripts, and most of them are not retryable).

**Comment by ramli01** [comment]
Yes exactly. I just continue klick upgrade button after fix the two errors.
I will try to restore database, fixed it first and run upgrade.
Thank you

**Comment by mabere** [comment]
Hi @jonasraoni. I have tried new upgrade using the restored database after fixing the errors. It stopped and shown the following error:
A database error has occurred: SQLSTATE[42S22]: Column not found: 1054 Unknown column 'at.assoc_type' in 'where clause' (SQL: select distinct `at`.`type_id` from `announcement_types` as `at` left join `journals` as `c` on `at`.`assoc_id` = `c`.`journal_id` where `c`.`journal_id` is null or `at`.`assoc_type` <> 256)

How it should be handled?

**Comment by jonasraoni** [comment]
Strange, your OJS 3.3 should have this field 😭
Using the initial database, what's the output for `SHOW CREATE TABLE announcement_types`?

**Comment by mabere** [comment]
Here the result:
CREATE TABLE `announcement_types` (
  `type_id` bigint(20) NOT NULL AUTO_INCREMENT,
  `context_id` bigint(20) NOT NULL,
  PRIMARY KEY (`type_id`),
  KEY `announcement_types_context_id` (`context_id`),
  CONSTRAINT `announcement_types_context_id_foreign` FOREIGN KEY (`context_id`) REFERENCES `journals` (`journal_id`) ON DELETE CASCADE
) ENGINE=InnoDB AUTO_INCREMENT=4 DEFAULT CHARSET=utf8 COLLATE=utf8_general_ci

**Comment by jonasraoni** [comment]
I see the `assoc_type` field is there, and there's no code to drop this field (at least, not before the place where it failed).
So the only thing that comes to my mind is that you retried to run the upgrade with the bad database.

**Comment by mabere** [comment]
Hi @jonasraoni. Thank you to keep responding and guiding me. Your are right. I rerun the failed upgrade database unintendedly.
Now, I have just try a new upgrade using restored database after fixing the duplication and local issues. And the upgrade stopped and show this error:
[Wed Jun 07 08:35:29.047347 2023] [php:notice] [pid 7584:tid 1888] [client ::1:50125] SQLSTATE[23000]: Integrity constraint violation: 1452 Cannot add or update a child row: a foreign key constraint fails (`dlej05`.`#sql-1e3c_e0`, CONSTRAINT `citations_publication` FOREIGN KEY (`publication_id`) REFERENCES `publications` (`publication_id`) ON DELETE CASCADE) (SQL: alter table `citations` add constraint `citations_publication` foreign key (`publication_id`) references `publications` (`publication_id`) on delete cascade), referer: http://localhost/dlej/index.php/index/install/upgrade

I try this query referring to #8929 : SELECT publication_id FROM citations WHERE publication_id NOT IN (SELECT publication_id FROM publications); in failed database and got one record for publication_id. But it brings no record when I the query in the restored database.
Is the query right or there something I missed?

**Comment by mabere** [comment]
Finally   @jonasraoni the upgrade is success. I upgrade using command line via git base.
I fixed the previous issue citations_publication by import the table _**citations**_ from the failed upgrade database (after deleting 1 publication_id record).

Here the upgrade process records:
$ php tools/upgrade.php upgrade
2023-06-07 05:10:41 [pre-install]
2023-06-07 05:10:41 [load: upgrade.xml]
2023-06-07 05:10:41 [version: 3.4.0.0]
2023-06-07 05:10:41 [code: Installer Installer::checkPhpVersion]
2023-06-07 05:10:41 [code: Installer Installer::installDefaultNavigationMenus]
WARNING: The NavigationMenu (ContextId: 1, Title: User Navigation Menu, Area: user) will be skipped because the speied area has already a NavigationMenu attached.
WARNING: The NavigationMenu (ContextId: 1, Title: Primary Navigation Menu, Area: primary) will be skipped because tspecified area has already a NavigationMenu attached.
WARNING: The NavigationMenu (ContextId: 0, Title: User Navigation Menu, Area: user) will be skipped because the specified area has already a NavigationMenu attached.
2023-06-07 05:10:41 [code: Installer Installer::migrateStaticPagesToNavigationMenuItems]
2023-06-07 05:10:41 [migration: APP\migration\upgrade\v3_4_0\PreflightCheckMigration]
2023-06-07 05:11:00 [Nulling non-existent genre_id 0 in submission_files.]
2023-06-07 05:11:01 [migration: PKP\migration\upgrade\v3_4_0\I8060_UpdateUserLocalesDefaultToEmptyArrayFromNull]
2023-06-07 05:11:03 [migration: PKP\migration\upgrade\v3_4_0\I7245_UpdateUserLocaleStringToParsableJsonString]
2023-06-07 05:11:09 [migration: APP\migration\upgrade\v3_4_0\I7129_IssueEntityDAORefactor]
2023-06-07 05:11:14 [migration: APP\migration\upgrade\OJSv3_3_1_i6872_UrlPathMigration]
2023-06-07 05:11:14 [migration: APP\migration\upgrade\v3_4_0\I6091_AddFilterNamespaces]
2023-06-07 05:11:16 [migration: APP\migration\upgrade\v3_4_0\I3573_AddPrimaryKeys]
2023-06-07 05:14:40 [migration: APP\migration\upgrade\v3_4_0\I6093_AddForeignKeys]
2023-06-07 05:19:13 [migration: APP\migration\upgrade\v3_4_0\MergeLocalesMigration]
2023-06-07 05:19:53 [migration: PKP\migration\upgrade\v3_4_0\FailedJobsMigration]
2023-06-07 05:19:54 [migration: PKP\migration\upgrade\v3_4_0\UpgradeMigration]
2023-06-07 05:20:05 [migration: APP\migration\upgrade\v3_4_0\I6807_SetLastModified]
2023-06-07 05:20:06 [migration: APP\migration\upgrade\v3_4_0\I4235_OAISetSpec]
2023-06-07 05:20:06 [migration: PKP\migration\upgrade\v3_4_0\I7167_RemoveDuplicatedUserSettingsAndDeprecatedFields]
2023-06-07 05:20:06 [migration: APP\migration\upgrade\v3_4_0\I7264_UpdateEmailTemplates]
2023-06-07 05:20:17 [migration: APP\migration\upgrade\v3_4_0\I7596_RemoveNonExpiring]
2023-06-07 05:20:17 [migration: APP\migration\upgrade\v3_4_0\I7014_DoiMigration]
2023-06-07 05:20:43 [migration: APP\migration\upgrade\v3_4_0\I7265_EditorialDecisions]
2023-06-07 05:20:49 [migration: PKP\migration\upgrade\v3_4_0\I7624_StrftimeDeprecation]
2023-06-07 05:20:50 [migration: PKP\migration\upgrade\v3_4_0\I7592_RemoveUnusedEmailTemplates]
2023-06-07 05:20:51 [migration: PKP\migration\upgrade\v3_4_0\I7126_Galleys]
2023-06-07 05:20:51 [migration: APP\migration\upgrade\v3_4_0\I7190_RemoveOrphanFilters]
2023-06-07 05:20:51 [migration: PKP\migration\upgrade\v3_4_0\I7190_UpdateFilters]
2023-06-07 05:20:52 [migration: PKP\migration\upgrade\v3_4_0\I6895_CreateNewInstitutionsTables]
2023-06-07 05:21:02 [migration: PKP\migration\upgrade\v3_4_0\I8073_RemoveNotesWithoutQueriesAndRelatedObjects]
2023-06-07 05:21:31 [A submission file that was attached to an orphaned note with ID 1 at journals/1/articles/15/submission/query/15-1-23-4-18-20200302.docx was successfully deleted.]
2023-06-07 05:21:31 [Removing orphaned note entry ID 1 with nonexistent query 1]
2023-06-07 05:21:32 [Removing orphaned note entry ID 19 with nonexistent query 18]
2023-06-07 05:21:32 [Removing orphaned note entry ID 20 with nonexistent query 19]
2023-06-07 05:21:32 [Removing orphaned note entry ID 28 with nonexistent query 26]
2023-06-07 05:21:32 [Removing orphaned note entry ID 29 with nonexistent query 27]
2023-06-07 05:21:32 [Removing orphaned note entry ID 30 with nonexistent query 28]
2023-06-07 05:21:32 [Removing orphaned note entry ID 31 with nonexistent query 29]
2023-06-07 05:21:33 [Removing orphaned note entry ID 61 with nonexistent query 49]
2023-06-07 05:21:33 [Removing orphaned note entry ID 62 with nonexistent query 50]
2023-06-07 05:21:33 [Removing orphaned note entry ID 99 with nonexistent query 80]
2023-06-07 05:21:33 [Removing orphaned note entry ID 106 with nonexistent query 86]
2023-06-07 05:21:33 [Removing orphaned note entry ID 125 with nonexistent query 100]
2023-06-07 05:21:33 [Removing orphaned note entry ID 145 with nonexistent query 110]
2023-06-07 05:21:33 [Removing orphaned note entry ID 146 with nonexistent query 111]
2023-06-07 05:21:33 [Removing orphaned note entry ID 147 with nonexistent query 112]
2023-06-07 05:21:34 [Removing orphaned note entry ID 151 with nonexistent query 115]
2023-06-07 05:21:34 [A submission file that was attached to an orphaned note with ID 152 at journals/1/articles/45/0e4080c8e9.docx was successfully deleted.]
2023-06-07 05:21:34 [Removing orphaned note entry ID 152 with nonexistent query 116]
2023-06-07 05:21:34 [A submission file that was attached to an orphaned note with ID 153 at journals/1/articles/45/143254c602.docx was successfully deleted.]
2023-06-07 05:21:34 [Removing orphaned note entry ID 153 with nonexistent query 117]
2023-06-07 05:21:35 [Removing orphaned note entry ID 156 with nonexistent query 120]
2023-06-07 05:21:35 [Removing orphaned note entry ID 157 with nonexistent query 121]
2023-06-07 05:21:35 [Removing orphaned note entry ID 158 with nonexistent query 122]
2023-06-07 05:21:35 [Removing orphaned note entry ID 159 with nonexistent query 123]
2023-06-07 05:21:35 [Removing orphaned note entry ID 163 with nonexistent query 127]
2023-06-07 05:21:35 [Removing orphaned note entry ID 171 with nonexistent query 133]
2023-06-07 05:21:35 [Removing orphaned note entry ID 174 with nonexistent query 135]
2023-06-07 05:21:35 [Removing orphaned note entry ID 175 with nonexistent query 136]
2023-06-07 05:21:35 [Removing orphaned note entry ID 190 with nonexistent query 146]
2023-06-07 05:21:35 [Removing orphaned note entry ID 192 with nonexistent query 147]
2023-06-07 05:21:35 [Removing orphaned note entry ID 194 with nonexistent query 149]
2023-06-07 05:21:35 [Removing orphaned note entry ID 203 with nonexistent query 158]
2023-06-07 05:21:35 [Removing orphaned note entry ID 207 with nonexistent query 162]
2023-06-07 05:21:35 [Removing orphaned note entry ID 211 with nonexistent query 165]
2023-06-07 05:21:35 [Removing orphaned note entry ID 215 with nonexistent query 167]
2023-06-07 05:21:36 [Removing orphaned note entry ID 217 with nonexistent query 168]
2023-06-07 05:21:36 [Removing orphaned note entry ID 223 with nonexistent query 174]
2023-06-07 05:21:37 [A submission file that was attached to an orphaned note with ID 224 at journals/1/articles/51/aed236fdc5.pdf was successfully deleted.]
2023-06-07 05:21:37 [Removing orphaned note entry ID 224 with nonexistent query 175]
2023-06-07 05:21:37 [Removing orphaned note entry ID 233 with nonexistent query 182]
2023-06-07 05:21:37 [Removing orphaned note entry ID 234 with nonexistent query 183]
2023-06-07 05:21:37 [Removing orphaned note entry ID 238 with nonexistent query 185]
2023-06-07 05:21:37 [Removing orphaned note entry ID 252 with nonexistent query 190]
2023-06-07 05:21:37 [Removing orphaned note entry ID 254 with nonexistent query 191]
2023-06-07 05:21:37 [Removing orphaned note entry ID 256 with nonexistent query 192]
2023-06-07 05:21:37 [Removing orphaned note entry ID 257 with nonexistent query 193]
2023-06-07 05:21:37 [Removing orphaned note entry ID 268 with nonexistent query 200]
2023-06-07 05:21:37 [Removing orphaned note entry ID 270 with nonexistent query 202]
2023-06-07 05:21:37 [Removing orphaned note entry ID 271 with nonexistent query 203]
2023-06-07 05:21:38 [Removing orphaned note entry ID 272 with nonexistent query 204]
2023-06-07 05:21:38 [Removing orphaned note entry ID 273 with nonexistent query 205]
2023-06-07 05:21:39 [Removing orphaned note entry ID 276 with nonexistent query 207]
2023-06-07 05:21:39 [A submission file that was attached to an orphaned note with ID 279 at journals/1/articles/53/45a636bfbf.docx was successfully deleted.]
2023-06-07 05:21:39 [Removing orphaned note entry ID 279 with nonexistent query 210]
2023-06-07 05:21:39 [Removing orphaned note entry ID 280 with nonexistent query 211]
2023-06-07 05:21:40 [A submission file that was attached to an orphaned note with ID 281 at journals/1/articles/53/45ac9a49f0.docx was successfully deleted.]
2023-06-07 05:21:40 [Removing orphaned note entry ID 281 with nonexistent query 212]
2023-06-07 05:21:40 [Removing orphaned note entry ID 285 with nonexistent query 215]
2023-06-07 05:21:40 [Removing orphaned note entry ID 295 with nonexistent query 222]
2023-06-07 05:21:40 [Removing orphaned note entry ID 296 with nonexistent query 223]
2023-06-07 05:21:40 [Removing orphaned note entry ID 322 with nonexistent query 236]
2023-06-07 05:21:40 [Removing orphaned note entry ID 323 with nonexistent query 237]
2023-06-07 05:21:40 [Removing orphaned note entry ID 332 with nonexistent query 241]
2023-06-07 05:21:40 [Removing orphaned note entry ID 333 with nonexistent query 242]
2023-06-07 05:21:40 [Removing orphaned note entry ID 334 with nonexistent query 243]
2023-06-07 05:21:40 [Removing orphaned note entry ID 335 with nonexistent query 244]
2023-06-07 05:21:40 [Removing orphaned note entry ID 336 with nonexistent query 245]
2023-06-07 05:21:41 [Removing orphaned note entry ID 341 with nonexistent query 247]
2023-06-07 05:21:41 [Removing orphaned note entry ID 342 with nonexistent query 248]
2023-06-07 05:21:41 [Removing orphaned note entry ID 350 with nonexistent query 252]
2023-06-07 05:21:41 [Removing orphaned note entry ID 356 with nonexistent query 256]
2023-06-07 05:21:41 [Removing orphaned note entry ID 358 with nonexistent query 258]
2023-06-07 05:21:41 [Removing orphaned note entry ID 363 with nonexistent query 263]
2023-06-07 05:21:41 [Removing orphaned note entry ID 364 with nonexistent query 264]
2023-06-07 05:21:41 [Removing orphaned note entry ID 365 with nonexistent query 265]
2023-06-07 05:21:42 [Removing orphaned note entry ID 368 with nonexistent query 267]
2023-06-07 05:21:42 [Removing orphaned note entry ID 370 with nonexistent query 268]
2023-06-07 05:21:42 [Removing orphaned note entry ID 379 with nonexistent query 275]
2023-06-07 05:21:43 [Removing orphaned note entry ID 380 with nonexistent query 276]
2023-06-07 05:21:43 [Removing orphaned note entry ID 382 with nonexistent query 278]
2023-06-07 05:21:43 [A submission file that was attached to an orphaned note with ID 383 at journals/1/articles/51/161d24421d.docx was successfully deleted.]
2023-06-07 05:21:43 [A submission file that was attached to an orphaned note with ID 383 at journals/1/articles/51/161f458ece.docx was successfully deleted.]
2023-06-07 05:21:43 [Removing orphaned note entry ID 383 with nonexistent query 279]
2023-06-07 05:21:43 [Removing orphaned note entry ID 385 with nonexistent query 281]
2023-06-07 05:21:44 [A submission file that was attached to an orphaned note with ID 388 at journals/1/articles/60/2bee40607f.docx was successfully deleted.]
2023-06-07 05:21:44 [Removing orphaned note entry ID 388 with nonexistent query 283]
2023-06-07 05:21:44 [Removing orphaned note entry ID 390 with nonexistent query 284]
2023-06-07 05:21:44 [A submission file that was attached to an orphaned note with ID 401 at journals/1/articles/51/7eff709328.docx was successfully deleted.]
2023-06-07 05:21:44 [Removing orphaned note entry ID 401 with nonexistent query 287]
2023-06-07 05:21:44 [Removing orphaned note entry ID 406 with nonexistent query 289]
2023-06-07 05:21:44 [Removing orphaned note entry ID 409 with nonexistent query 292]
2023-06-07 05:21:44 [Removing orphaned note entry ID 417 with nonexistent query 300]
2023-06-07 05:21:45 [Removing orphaned note entry ID 430 with nonexistent query 313]
2023-06-07 05:21:45 [Removing orphaned note entry ID 431 with nonexistent query 314]
2023-06-07 05:21:45 [Removing orphaned note entry ID 434 with nonexistent query 315]
2023-06-07 05:21:45 [Removing orphaned note entry ID 436 with nonexistent query 316]
2023-06-07 05:21:45 [Removing orphaned note entry ID 449 with nonexistent query 326]
2023-06-07 05:21:45 [Removing orphaned note entry ID 454 with nonexistent query 330]
2023-06-07 05:21:45 [Removing orphaned note entry ID 457 with nonexistent query 333]
2023-06-07 05:21:45 [Removing orphaned note entry ID 458 with nonexistent query 334]
2023-06-07 05:21:46 [Removing orphaned note entry ID 467 with nonexistent query 338]
2023-06-07 05:21:46 [Removing orphaned note entry ID 475 with nonexistent query 342]
2023-06-07 05:21:46 [Removing orphaned note entry ID 487 with nonexistent query 346]
2023-06-07 05:21:46 [Removing orphaned note entry ID 490 with nonexistent query 349]
2023-06-07 05:21:46 [Removing orphaned note entry ID 515 with nonexistent query 373]
2023-06-07 05:21:46 [A submission file that was attached to an orphaned note with ID 524 at journals/1/articles/87/9280f05dd2.docx was successfully deleted.]
2023-06-07 05:21:46 [Removing orphaned note entry ID 524 with nonexistent query 382]
2023-06-07 05:21:46 [Removing orphaned note entry ID 533 with nonexistent query 389]
2023-06-07 05:21:47 [Removing orphaned note entry ID 534 with nonexistent query 390]
2023-06-07 05:21:47 [Removing orphaned note entry ID 536 with nonexistent query 392]
2023-06-07 05:21:47 [Removing orphaned note entry ID 542 with nonexistent query 398]
2023-06-07 05:21:47 [Removing orphaned note entry ID 547 with nonexistent query 402]
2023-06-07 05:21:47 [migration: APP\migration\upgrade\v3_4_0\I6895_Institutions]
2023-06-07 05:21:52 [migration: APP\migration\upgrade\v3_4_0\I6782_OrphanedMetrics]
2023-06-07 05:21:56 [Removing stats for context 2 because no context with that ID could be found.]
2023-06-07 05:21:57 [migration: PKP\migration\upgrade\v3_4_0\I6782_UsageStatsSettings]
2023-06-07 05:21:58 [migration: APP\migration\upgrade\v3_4_0\I6782_CreateNewMetricsTables]
2023-06-07 05:23:44 [migration: APP\migration\upgrade\v3_4_0\I6782_MetricsContext]
2023-06-07 05:23:46 [migration: APP\migration\upgrade\v3_4_0\I6782_MetricsIssue]
2023-06-07 05:23:48 [migration: APP\migration\upgrade\v3_4_0\I6782_MetricsSubmission]
2023-06-07 05:23:58 [migration: APP\migration\upgrade\v3_4_0\I6782_MetricsGeo]
2023-06-07 05:23:58 [migration: APP\migration\upgrade\v3_4_0\I6782_CleanOldMetrics]
2023-06-07 05:24:13 [migration: APP\migration\upgrade\v3_4_0\I6782_RemovePlugins]
2023-06-07 05:24:15 [migration: PKP\migration\upgrade\v3_4_0\I7286_BatchesMigration]
2023-06-07 05:24:17 [migration: APP\migration\upgrade\v3_4_0\I8151_ExtendSettingValues]
2023-06-07 05:24:31 [migration: PKP\migration\upgrade\v3_4_0\I8151_ExtendSettingValues]
2023-06-07 05:25:40 [migration: PKP\migration\upgrade\v3_4_0\I2890_EmailTemplatesVarcharLengthUpdate]
2023-06-07 05:25:48 [migration: PKP\migration\upgrade\v3_4_0\I2890_AddSetNullForOnDeleteToReviewRoundIdForeign]
2023-06-07 05:25:49 [migration: APP\migration\upgrade\v3_4_0\I7901_Duplicate_OAI_IDs]
2023-06-07 05:25:50 [migration: PKP\migration\upgrade\v3_4_0\I7366_UpdateUserAPIKeySettings]
2023-06-07 05:25:50 [migration: PKP\migration\upgrade\v3_4_0\I8093_UpdateUserGroupRelationTablesFK]
2023-06-07 05:25:56 [migration: APP\migration\upgrade\v3_4_0\I7725_DecisionConstantsUpdate]
2023-06-07 05:25:57 [migration: PKP\migration\upgrade\v3_4_0\I4789_AddReviewerRequestResentColumns]
2023-06-07 05:25:57 [migration: APP\migration\upgrade\v3_4_0\I7796_UpdateCrossrefSchema]
2023-06-07 05:25:57 [migration: PKP\migration\upgrade\v3_4_0\I7287_RemoveEmailTemplatesDefault]
2023-06-07 05:25:58 [migration: APP\migration\upgrade\v3_4_0\I5716_EmailTemplateAssignments]
2023-06-07 05:26:12 [migration: APP\migration\upgrade\v3_4_0\InstallEmailTemplates]
2023-06-07 05:26:14 [migration: PKP\migration\upgrade\v3_4_0\I7874_NotificationMetadataModifiedRemove]
2023-06-07 05:26:14 [migration: APP\migration\upgrade\v3_4_0\I7191_EditorAssignments]
2023-06-07 05:26:20 [migration: APP\migration\upgrade\v3_4_0\I7191_InstallSubmissionHelpDefaults]
2023-06-07 05:26:20 [migration: APP\migration\upgrade\v3_4_0\I7191_SubmissionChecklistMigration]
2023-06-07 05:26:21 [migration: PKP\migration\upgrade\v3_4_0\I7191_SubmissionProgressType]
2023-06-07 05:26:23 [migration: PKP\migration\upgrade\v3_4_0\I7191_ResubscribeSubeditors]
2023-06-07 05:26:23 [migration: APP\migration\upgrade\v3_4_0\I8027_DoiVersioning]
2023-06-07 05:26:23 [migration: APP\migration\upgrade\v3_4_0\I6306_EnableCategories]
2023-06-07 05:26:23 [migration: PKP\migration\upgrade\v3_4_0\I8592_SiteNotificationSubscriptions]
2023-06-07 05:26:26 [migration: APP\migration\upgrade\v3_4_0\I6241_RequiredGenres]
2023-06-07 05:26:26 [migration: APP\migration\upgrade\v3_4_0\I7128_SectionEntityDAORefactor]
2023-06-07 05:26:26 [migration: APP\migration\upgrade\v3_4_0\I7513_DoiSettings]
2023-06-07 05:26:27 [migration: PKP\migration\upgrade\v3_4_0\I7486_RenameUnconsideredColumnToConsidered]
2023-06-07 05:26:29 [migration: PKP\migration\upgrade\v3_4_0\I7486_RemoveItemViewsTable]
2023-06-07 05:26:29 [migration: PKP\migration\upgrade\v3_4_0\I8737_SectionEditorsUniqueIndexUpdate]
2023-06-07 05:26:30 [migration: PKP\migration\upgrade\v3_4_0\I8508_ConvertCurrentLogFile]
Process could not be started: The system cannot find the path specified.
2023-06-07 05:26:34 [migration: PKP\migration\upgrade\v3_4_0\I8866_DispatchRegionCodesFixingJobs]
2023-06-07 05:26:34 [data: dbscripts/xml/upgrade/3.4.0_preupdate_email_templates.xml]
2023-06-07 05:26:34 [note: docs/release-notes/README-3.4.0]
2023-06-07 05:26:34 [migration: PKP\migration\upgrade\v3_4_0\I7249_UpdateUsersUniqueIndex]
2023-06-07 05:26:34 [code: Installer Installer::addPluginVersions]
2023-06-07 05:26:37 [post-install]

Release Notes
----------------------------------------
OJS 3.4.0rc2 Release Notes
Git tag: 3_4_0rc2

Release date: March 29, 2023
============================

=========================================================================
THIS IS A PRE-RELEASE VERSION OF OJS AND NOT INTENDED FOR PRODUCTION USE.
=========================================================================

Configuration Changes
---------------------

See config.TEMPLATE.inc.php for a description and examples of all supported configuration parameters.

- The date/time formats in config.inc.php should be converted from the format
  specified in https://www.php.net/manual/en/function.strftime.php to the format
  specified in https://www.php.net/manual/en/datetime.format.php. If you're not
  sure how, use the values below. The following settings are affected:
    date_format_short = "Y-m-d"
    date_format_long = "F n, Y"
    datetime_format_short = "Y-m-d h:i A"
    datetime_format_long = "F n, Y - h:i A"
    time_format = "h:i A"
  The old formats will continue to work for now but are deprecated.

- An allowed_hosts option was added to protect against HOST injection attacks.
  We recommend configuring this setting! See "Recommended Configuration" in docs/README.md for details. (https://github.com/pkp/pkp-lib/issues/7649)

- The captcha.captcha_on_login option for displaying a captcha challenge in the login interface has been added (default: on)
- The setting i18n.client_charset was removed. It used to support two options "ISO-8859-1" (LATIN1) and "UTF-8", no we're UTF-8 only.
- The setting i18n.locale will likely need to be updated because locale codes have been standardized. For example, en__US" has become "en".
    Check the `locale/` directory for a list of available locale codes.

New config.inc.php section for queues is added, with the following parameters:
    - default_connection (default value: database), being the default queue driver to be used;
    - default_queue (default value: queue), being the default queue to be used if someone forgot to associate a queue during Queue::push()
    - disable_jobs_run_at_shutdown (default value: off), being a flag to disable the jobs to run at php shutdown

- The setting general.session_samesite (default value: Lax) has been added to control the "SameSite" setting for the session cookie.
- The setting security.force_ssl, when enabled, will also set the "Secure" flag for the session cookie.

New Features
------------

Bug Fixes
---------

Successfully upgraded to version 3.4.0.0

**Comment by mabere** [comment]
I do not know whether it is a right way. But so far, the ojs web works fine. I would fully test the features to check if there is still an issue or no.

**Comment by jonasraoni** [comment]
Thanks for the help, so I'll close this issue.

About the problem that you had with orphaned data (` Cannot add or update a child row: a foreign key constraint fails`), this is going to be fixed by: https://github.com/pkp/pkp-lib/issues/9069.

If you find something broken during your tests (e.g. Error 500), you might try to search for a similar problem here: https://github.com/pkp/pkp-lib/issues

**Comment by jonasraoni** [comment]
Hi @mabere! OJS 3.4 was released today, so it would be great to hear your feedback about the upgrade experience before we start the marketing/promotion.
Besides the corrections, we also added some automatic fixes, so the process should be less problematic.

**Comment by mabere** [comment]
Great news.I've just test for fresh installation and it run with no issue and am just exploring its features.
I will reach you out if found any issue.
Thanks.

    On Saturday, 10 June 2023 at 14:09:00 GMT+8, Jonas Raoni Soares da Silva ***@***.***> wrote:  
 
 


Hi @mabere! OJS 3.4 was released today, so it would be great to hear your feedback about the upgrade experience before we start the marketing/promotion.
Besides the corrections, we also added some automatic fixes, so the process should be less problematic.

**Comment by jonasraoni** [comment]
Great, thanks for the fast response 😁
